controllers/managedocs_controller.go

```diff
@@ -1030,6 +1033,19 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			}
 		}
 
+		// getting clusterID
+		clusterVersion := &configv1.ClusterVersion{}
+		err1 := r.UnrestrictedClient.Get(context.Background(), types.NamespacedName{Name: "version"}, clusterVersion)
+
+		if err1 != nil {
+			r.Log.Info(err1.Error())
+		} else {
+			r.Log.Info("Cluster ID - " + string(clusterVersion.Spec.ClusterID))
+
+			clusterID := string(clusterVersion.Spec.ClusterID)
+			desired.Spec.ExternalLabels["clusterid"] = clusterID
+		}
```

Comment:
```suggestion
		clusterVersion := &configv1.ClusterVersion{}
		if err := r.UnrestrictedClient.Get(context.Background(), types.NamespacedName{Name: "version"}, clusterVersion); err != nil {
			return err
		}
		clusterID := string(clusterVersion.Spec.ClusterID)
		desired.Spec.ExternalLabels["clusterid"] = clusterID
```
You should use `unrestrictedGet` function.
I don't think we need to log error and clusterID so I have removed it in the suggestion.

Reply:
You can even return a meaningful error message along with err(use fmt.Errorf).